### PR TITLE
Pricing page rework: Add HotJar analytics script

### PR DESCRIPTION
**Pull request by elliottprogrammer**
#### Proposed Changes

This PR adds the [HotJar analytics script](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/analytics/README_HotJar.md) into the new pricing page, so we can closely measure the impact of the new page.

See: P2 "Measuring the impact of the new pricing page/structure" -->  p1HpG7-hEn-p2

Asana Task: 1202796695664022-as-1202898236955395/f

**Implementation notes:**

- It loads the HotJar script on routes `cloud.jetpack.com/pricing/..` and `wordpress.com/jetpack/connect/plans/:site/..` (the Jetpack plugin post-connection route)
- This PR _also_ enables the `config( 'hotjar_enabled')` in the Jetpack Cloud environment for staging & production.  HotJar was already enabled for wordpress.com, so no need to enable it there.

#### Testing Instructions

* Run `git fetch && git checkout add/hotjar-jetpack-cloud-pricing`
  * Open file `config/jetpack-cloud-development.json`, and set the `"hotjar_enabled"` property to `true`.
  * Run `yarn start` in one command line tab/window, and also run `yarn start-jetpack-cloud-p' in another command line tab/window. (This runs Calypso blue and Calypso green concurrently)
  * Go to `cloud.jetpack.com:3001/pricing?flags=jetpack/pricing-page-rework-v1`.
  * Open dev tools console and enter `localStorage.debug = 'calypso:analytics:hotjar'; and press enter.
  * Reload the page and verify in the console you see the debug statement: `calypso:analytics:hotjar Loading HotJar script`. (You will also see an error that "HotJar only works over HTTPS", but that's ok for our purposes here. We are just testing that the script is trying to load.)
  * Next, visit `calypso.localhost:3000/jetpack/connect/plans/:site?flags=jetpack/pricing-page-rework-v1` (where `:site` is a connected Jetpack site you own (or if you don't already have a connected Jetpack site, create a JN site and connect it)). Users are redirected to this page right after Jetpack plugin connection.
  * Again open dev tools console and enter `localStorage.debug = 'calypso:analytics:hotjar'; and press enter.
  * Reload the page and verify in the console you see the debug statement: `calypso:analytics:hotjar Loading HotJar script`.
  * Now go to `http://calypso.localhost:3000/plans/site?flags=jetpack/pricing-page-rework-v1` and verify that the HotJar script is **not** loading for this page (`wordpress.com/plans/` page). (There should **not** be a HotJar debug statement in the console.)
  * Now also go to `http://calypso.localhost:3000/jetpack/connect/store?flags=jetpack/pricing-page-rework-v1`and verify that the HotJar script is **not** loading for this page either.
  

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202898236955395

### CODE REVIEW

**Review by jkguidaven — APPROVED** [review]
Codes LGTM 💯 . Followed the testing instructions and it is working as expected.

 I'm curious why we are not tracking `wordpress.com/plans/` and `wordpress.com/jetpack/connect/store`?

**Comment by manzoorwanijk** [comment]
> I'm curious why we are not tracking `wordpress.com/plans/` and `wordpress.com/jetpack/connect/store`?

@jkguidaven as already mentioned in PR description:

> HotJar was already enabled for wordpress.com, so no need to enable it there.

**Review by manzoorwanijk — APPROVED** [review]
This works as expected

**Comment by elliottprogrammer** [comment]
> I'm curious why we are not tracking wordpress.com/plans/ and wordpress.com/jetpack/connect/store?

@manzoorwanijk , @jkguidaven , Yes good question, thank you. I probably should have explained this in the description/instructions. My thinking was this:
Our HotJar plan is quite limited in the number of user screen recordings it will capture (approx 3000/mo?, and it's already being used in other places also.), so we only want to track the most utilized/relevant **Jetpack** purchase flows. I believe the most common/relevant purchase flows originate from Jetpack.com and from the plugin itself. From Jetpack.com, users are directed to the store at `cloud.Jetpack.com/pricing/` page. Then also right after Jetpack plugin connection, users are directed to the store at `wordpress.com/jetpack/connect/plans/:site`.  Then also directly from the Jetpack plugin, users can purchase a product/plan by clicking the "Plans" link in the dashboard which directs them to the store at `cloud.Jetpack.com/pricing/:site`.
I don't think it is very common for Jetpack users to end up on `wordpress.com/plans/:site` to purchase plans (I'm not even sure how they would arrive there, other than if they are already on wordpress.com managing a wpcom site or something, and then decide to select one of their Jetpack sites and then go to Jetpack -> Plans, in Calypso blue, and although it's possible to purchase (for Jetpack) this way, I wouldn't think it is very common.)
And the same goes for `wordpress.com/Jetpack/connect/store/`. In fact I'm not sure that route is even used anymore. I think it used to be used within an automated  "jetpack connect" flow, but now since licensing has been integrated, that flow is no longer in use (I think, AFAIK).
So that is my thinking, but if I am missing something or you see something that I don't, please comment! 🙂

File: config/jetpack-cloud-development.json [hunk 1]
```diff
@@ -83,5 +83,5 @@
 	"restricted_me_access": false,
 	"theme_color": "#2fb41f",
 	"google_analytics_key": "UA-52447-43",
-	"hotjar_enabled": false
+	"hotjar_enabled": true
```
Comment (elliottprogrammer) [inline config/jetpack-cloud-development.json:86]: HotJar is previously already enabled in **all** Calypso blue environments (wordpress.com), So for consistency, I'll enable HotJar in **all** Calypso green environments (cloud.jetpack.com) too.

**Pull request merged**